pennylane/io.py

```diff
@@ -122,11 +109,177 @@ def from_qiskit(quantum_circuit, measurements=None):
     Returns:
         function: the PennyLane template created based on the ``QuantumCircuit`` object
 
-    .. details::
-        :title: Usage Details
+    **Example:**
+
+    .. code-block:: python
+
+        import pennylane as qml
+        from qiskit import QuantumCircuit
+
+        qc = QuantumCircuit(2, 2)
+        qc.rx(0.785, 0)
+        qc.ry(1.57, 1)
+
+        my_qfunc = qml.from_qiskit(qc)
+
+    The ``my_qfunc`` function can now be used within QNodes, as a two-wire quantum
+    template. We can also pass ``wires`` when calling the returned template to define
+    which device wires it should operate on. If no wires are passed, it will default
+    to sequential wire labels starting at 0.
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit")
+
+        @qml.qnode(dev)
+        def circuit():
+            my_qfunc(wires=["a", "b"])
+            return qml.expval(qml.Z("a")), qml.var(qml.Z("b"))
+
+    >>> circuit()
+    (tensor(0.70738827, requires_grad=True),
+    tensor(0.99999937, requires_grad=True))
+
+    The measurements can also be passed directly to the function when creating the
+    qfunc, making it possible to create a PennyLane circuit with ``qml.QNode`:
+
+    >>> measurements = [qml.expval(qml.Z(0)), qml.var(qml.Z(1))]
+    >>> circuit = qml.QNode(qml.from_qiskit(qc, measurements), dev)
+    >>> circuit()
+    (tensor(0.70738827, requires_grad=True),
+    tensor(0.99999937, requires_grad=True))
+
+    .. note::
 
         The ``measurement`` keyword allows one to add a list of PennyLane measurements
-        that will override the terminal measurements present in their ``QuantumCircuit``.
+        that will **override** the terminal measurements present in the ``QuantumCircuit``,
+        so that they are not performed before the operations specified in ``measurements``.
+        Converting a QuantumCircuit with ``measurements`` set will create a quanutm function
+        that does not return final or mid-circuit measurement values. See Usage Details below
+        for more information on how measurements defined on the QuantumCircuit are returned if
+        ``measurements=None``.
+
+    If an existing ``QuantumCircuit`` already ends with final measurements, the QNode can return
+    the expectation values of those final measurements directly:
```

Comment:
```suggestion
    If an existing ``QuantumCircuit`` already contains measurements, ``from_qiskit``
    will return those measurements, which can be used, e.g., for conditioning with
    :func:`qml.cond() <~.cond>`, or simply included directly within the QNode's return:
```

Reply:
@trbromley Looks good although I'm curious using why we write QNode instead of `QNode` here.